### PR TITLE
Finish implementing RFC 6265 cookies.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
Interestingly, the tests found a difference between JDK7 and JDK8. On JDK7 `example.com.` is converted to `example.com` by the IDN mechanism. (We don’t want this.)

So big win for having Travis running multiple versions of the JDK.

**Comment by swankjesse** [comment]
Failure is https://github.com/square/okhttp/issues/2165

File: okhttp/src/main/java/okhttp3/internal/Util.java [hunk 1]
```diff
@@ -371,4 +387,50 @@ public static int delimiterOffset(String input, int pos, int limit, char delimit
...
+    for (int i = 0; i < hostnameAscii.length(); i++) {
+      char c = hostnameAscii.charAt(i);
+      // The WHATWG Host parsing rules accepts some character codes which are invalid by
+      // definition for OkHttp's host header checks (and the WHATWG Host syntax definition). Here
+      // we rule out characters that would cause problems in host headers.
+      if (c <= '\u001f' || c >= '\u007f') {
+        return true;
+      }
+      // Check for the characters mentioned in the WHATWG Host parsing spec:
+      // U+0000, U+0009, U+000A, U+000D, U+0020, "#", "%", "/", ":", "?", "@", "[", "\", and "]"
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/internal/Util.java]: Not seeing the first four items from this list in the check.

Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/internal/Util.java]: Though this existed before...

Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/Util.java]: (the first four are in `0x0000...0x001f` above.)

Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/Util.java]: These two methods moved from HttpUrl.

**Comment by JakeWharton** [comment]
:+1:

**Pull request merged**